docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +347,271 @@ recursion was explicitly requested.
+  if x {
+    while false {
+      assert y >= 0;          // --         Assertion
+      assert {:focus} y <= 2; // --         Assertion
+      y := 2;
+      assert y == 2;          // --         Assertion
+    }
+  } else {
+    assert y == 1;            // Assertion  --
+  }
+  assert y == 1;              // Assertion  --
+  if !x {
+    assert y >= 1;            // Assertion  --
+  } else {
+    assert y <= 1;            // Assertion  --
+  }
+}
+```
+
+### 22.3.2. `{:split_here}` {#sec-split_here}
+`assert {:split_here} X;` splits verification into two [assertion batches](#sec-assertion-batches).
+It verifies the code leading to this point (excluded) in a first assertion batch,
+and the code leading from this point (included) to the next `{:split_here}` or until the end in a second assertion batch.
+It might help with timeouts.
+
+Here is one example, where `--` in the comments stands for `Assumption`:
+```dafny
+method doSplitHere(x: bool) returns (y: int) {
+  y := 1;                      // Batch 1    Batch 2     Batch 3
+  assert y >= 0;               // Assertion  --          --
+  if x {
+    assert y <= 1;             // Assertion  --          --
+    assert {:split_here} true; // --         Assertion   --
+    assert y <= 2;             // --         Assertion   --
+    assert {:split_here} true; // --         --          Assertion
+    if x {
+      assert y == 1;           // --         --          Assertion
+    } else {
+      assert y >= 1;           // --         --          Assertion
+    }
+  } else {
+    assert y <= 3;             // Assertion  --          --
+  }
+  assert y >= -1;              // Assertion  --          --
+}
+```
+
+### 22.3.3. `{:subsumption n}`
+Overrides the `/subsumption` command-line setting for this assertion.
+
+## 22.4. Attributes on variable declarations
+
+### 22.4.1. `{:assumption}`
+This attribute can only be placed on a local ghost bool
+variable of a method. Its declaration cannot have a rhs, but it is
+allowed to participate as the lhs of exactly one assignment of the
+form: `b := b && expr;`. Such a variable declaration translates in the
+Boogie output to a declaration followed by an `assume b` command.
+See [@LeinoWuestholz2015], Section 3, for example uses of the `{:assumption}`
+attribute in Boogie.
 
+### 22.4.2. `{:existential}`
+Marks a global Boolean variable as existentially quantified. If
+used in combination with option `/contractInfer`, Boogie will check
+whether there exists a Boolean assignment to the existentials
+that makes all verification conditions valid.  Without option
+`/contractInfer` the attribute is ignored.
 
-## 22.2. Boogie Attributes
-Use the Boogie "/attrHelp" option to get the list of attributes
-that Boogie recognizes and their meaning. Here is the output at
-the time of this writing. Dafny passes attributes that have
-been specified to Boogie.
+## 22.5. Attributes on quantifier expressions (forall, exists)
+
+### 22.5.1. `{:heapQuantifier}`
```

Comment:
This one is meaningless to me as a Dafny programmer AFAICT, so I'd skip it as well.

Reply:
I can't skip it because it has some meaning, and some specific translation to Boogie, so even if I don't know how to use it, I have no reason to believe it can't be useful for someone dealing with the heap.

Reply:
Fair enough, I realize that lifting the content up here rather than just saying "here's the dump of attributes Boogie understands, IDK you figure it out" :) means places where we haven't mapped things back to Dafny are more obvious. I'm satisfied we're making forward progress with this at least.